jerry-core/include/jerryscript-port.h

```diff
@@ -20,6 +20,32 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#if defined(_MSC_VER)
+#include <malloc.h>
+#endif
+
+#if defined(_MSC_VER)
+#define jerry_alloca(name, type, size) type *name = (type*)(alloca(sizeof(type)*size))
+#else
+#define jerry_alloca(name, type, size) type name[size]
+#endif
+
+#ifdef _MSC_VER
+#define __attr_unused___
+#define __attr_used___
+#ifndef __noreturn
+#define __noreturn __declspec(noreturn)
+#endif
+#define __format_type
+#else
+#define __attr_unused___ __attribute__((unused))
+#define __attr_used___ __attribute__((used))
+#ifndef __noreturn
+#define __noreturn __attribute__((noreturn))
+#endif
+#define __format_type __attribute__((format(printf, 2, 3)))
+#endif
```

Comment:
I see what's happening here and I also see why. I think it's not right this way, but my problem is that I cannot come up with a good alternative suggestion right now. The issue with this change is that jerryscript-port.h is there to define an API for port implementors to implement and for jerry-core to rely on. It's not to define an API that extensions, jerry-core users, etc. can use. At least that's how it used to be, and it was cleaner that way IMO.

But first things first: some definitions above can certainly be removed if jerry-libc and -libm are disabled for MSVC, let's see what remains.

PS: `jerry_alloca` should be all caps, provided that it remains.

PPS: `__format_type` should be something like `__attr_format___(...) __attribute__((format(__VA_ARGS__)))`. Only `__noreturn` deviates from the `__attr_xxx___` pattern and we should not introduce mor deviations IMO.

PPPS: Now I see why you wrote before that `__attr_unused___` comes from jerryscript-port.h. But I still think it shouldn't and that there should be a better/nicer way.

Reply:
1. jerry_alloca -> JERRY_ALLOCA ?
2. Define __attribute__ to empty macro? is this an acceptable idea?
PPPS: Looking for better suggestion, I am still not familier with jerry coding stardand.

Reply:
1. yes
2. That might work. It's quite a trick. It could even be done from cmake files, adding a `__attribute__(x)=` to the compile definitions (or to `DEFINES_JERRY` perhaps).

Reply:
@akosthekiss if the jerry api does not use __attribute__ then, in CMake should be fine.

Reply:
Fixesd